compiler-core/src/build/package_compiler.rs

```diff
@@ -29,13 +29,26 @@ use camino::{Utf8Path, Utf8PathBuf};
 
 use super::{ErlangAppCodegenConfiguration, TargetCodegenConfiguration, Telemetry};
 
+#[derive(Copy, Clone, Debug)]
+/// A flag that can be used to tell the package compiler to drop all the modules
+/// before performing code generation.
+/// For example we can use this to compile and run the dependencies of a project
+/// while ignoring the root package.
+pub enum ModulesCompilation {
+    /// Do not ignore the package's modules.
+    CompileAll,
+    /// Drop all the modules of a project before performing code generation.
+    IgnoreModules,
+}
+
```

Comment:
Could remove the repetition of "compile" and "modules" in the names here?
```suggestion
pub enum CompileModules {
    /// Do not ignore the package's modules.
    All,
    /// Drop all the modules of a project before performing code generation.
    None,
}

```